js/src/controls/SearchControl.js

```diff
@@ -19,22 +19,43 @@ export class LeafletSearchControlModel extends control.LeafletControlModel {
       zoom:10,
       animate_location:false,
       marker: null,
+      layer: null,
     };
   }
 }
 
 LeafletSearchControlModel.serializers = {
   ...control.LeafletControlModel.serializers,
-  marker: { deserialize: widgets.unpack_models }
+  marker: { deserialize: widgets.unpack_models },
+  layer: { deserialize: widgets.unpack_models },
 };
 
 
 export class LeafletSearchControlView extends control.LeafletControlView {
   create_obj() {
-    return this.create_child_view(this.model.get('marker')).then((view) => {
-      let options = this.get_options();
-      options.marker = view.obj;
-      this.obj = L.control.search(options);
-    });
+    if (this.model.get('layer') !== null) {
+        return this.create_child_view(this.model.get('layer')).then((layer_view) => {
+            let options = this.get_options();
+            options.layer = layer_view.obj;
+            options.marker = false;
+            this.obj = L.control.search(options);
+        });
```

Comment:
It looks like your implementation makes it impossible to set the `marker` and the `layer` attributes at the same time.

What about something like this for the `create_obj` method:
```javascript
const layer = this.model.get('layer');
const marker = this.model.get('marker');

const layer_promise = layer !== null ? this.create_child_view(layer) : Promise.resolve(null);
const marker_promise = marker !== null ? this.create_child_view(marker) : Promise.resolve(null);

return Promise.all([layer_promise, marker_promise]).then((layer_view, marker_view) => {
    const options = this.get_options();
    options.layer = layer_view !== null ? layer_view.obj : false;
    options.marker = marker_view !== null ? marker_view.obj : false;
    this.obj = L.control.search(options);
});
```

I did not try it locally so you might need to change one or two things.

Reply:
I will try this.

Reply:
You could indeed do that if the user did not define a Marker? But if the user defined a Marker I guess they will expect it to be used.

Reply:
Sure, I will try the changes you suggested.

Reply:
@martinRenou - updated the code as suggested by you.